spark/src/main/scala/geotrellis/spark/io/file/FileLayerDeleter.scala

```diff
@@ -20,43 +20,35 @@ import geotrellis.spark._
 import geotrellis.spark.io._
 import geotrellis.spark.io.AttributeStore.Fields
 import geotrellis.spark.io.index._
+import geotrellis.util.LazyLogging
 
 import spray.json.JsonFormat
 import org.apache.avro.Schema
 
 import scala.reflect.ClassTag
 import java.io.File
 
-object FileLayerDeleter {
-  def apply(attributeStore: FileAttributeStore): LayerDeleter[LayerId] =
-    new LayerDeleter[LayerId] {
-      def delete(layerId: LayerId): Unit = {
-        // Read the metadata file out.
-        val header =
-          try {
-            attributeStore.readHeader[FileLayerHeader](layerId)
-          } catch {
-            case e: AttributeNotFoundError =>
-              throw new LayerNotFoundError(layerId).initCause(e)
-          }
-
-        // Delete the attributes
-        for((_, file) <- attributeStore.attributeFiles(layerId)) {
-          file.delete()
-        }
-
-        // Delete the elements
-        val sourceLayerPath = new File(attributeStore.catalogPath, header.path)
-        if(sourceLayerPath.exists()) {
-          sourceLayerPath
-            .listFiles()
-            .foreach(_.delete())
-
-          sourceLayerPath.delete
-        }
+class FileLayerDeleter(val attributeStore: FileAttributeStore) extends LazyLogging with LayerDeleter[LayerId] {
```

Comment:
This is worth taking a close look at - I don't think `new`ing a trait inside an apply is a good idea for readability.